### PR TITLE
Developer onboarding resources

### PR DESCRIPTION
**Pull request by dasmanas**
## Description

Closes pyrsia/pyrsia#1336

This PR adds better onboarding documentation. This includes a ready list of libraries that are being used and related resources to have better onboarding experience for developers.

## PR Checklist

- [x] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [x] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/community/get_involved/good_pr.md)
- [x] I've included a good title and brief description along with how to review them.
- [x] I've linked any associated an [issue](https://github.com/pyrsia/pyrsia/issues).

### Code Contributions

- [x] I've built the code `cargo build --all-targets` successfully.
- [x] I've run the unit tests `cargo test --workspace` and all tests pass.
- [x] I've run the [integrations tests](https://github.com/pyrsia/pyrsia-integration-tests#how-to-set-up-and-run-the-tests) and all tests passes.

### CODE REVIEW

File: docs/community/get_involved/developer_onboarding.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Developer Onboarding
+
+If you are interested to contribute in Pyrsia project, here are a few important resources which might helpful on your onboarding journey.
```
Comment (betarelease) [inline docs/community/get_involved/developer_onboarding.md:3]:
```suggestion
If you are interested to contribute in Pyrsia project, here are a few important resources which might be helpful on your onboarding journey.
```

File: docs/community/get_involved/developer_onboarding.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+## Join, Subscribe & Access
+
+Request membership access to join [googlegroup for Pyrsia](https://groups.google.com/g/pyrsia). This will allow you to
+receive emails and calendar invites send to Pyrsia Googlegroup.
+
+We collaborate different project related artifacts using Google Drive's shared directory
```
Comment (betarelease) [inline docs/community/get_involved/developer_onboarding.md:10]:
```suggestion
We collaborate on the project using Google Drive's shared directory
```

File: docs/community/get_involved/developer_onboarding.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+- Chapter by Chapter [The Rust Lang Book Video Playlist](https://www.youtube.com/playlist?list=PLai5B987bZ9CoVR-QEIN9foz4QCJ0H2Y8)
+- [The Cargo Book](https://doc.rust-lang.org/cargo/index.html) - Rust Package Manager
+- A collection of runnable example at [Rust By Example](https://doc.rust-lang.org/stable/rust-by-example/)
+- In Pyrsia we use Asynchronous Programing in quite a few places. You would find important concepts at [Asynchronous Programming in Rust](https://rust-lang.github.io/async-book/)
+
+### Third Party Frameworks & Libraries
+
+To build Pyrsia's decentralized network we made some architectural design decisions for **_peer to peer_**
+communications. This requires us to use a few protocols and their third party implementations. Here is a list of few
+major of them along with useful links to relevant documents.
```
Comment (betarelease) [inline docs/community/get_involved/developer_onboarding.md:39]:
```suggestion
of them along with useful links to relevant documents.
```

File: docs/community/get_involved/developer_onboarding.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+- We are profoundly using libp2p in Pyrsia project development. It is recommended to go through some more details from
```
Comment (betarelease) [inline docs/community/get_involved/developer_onboarding.md:41]:
```suggestion
Pyrsia uses libp2p library for building the decentralized network
```

**Review by betarelease — APPROVED** [review]
I left some suggestions but overall great start.

**Comment by dasmanas** [comment]
> 



> I left some suggestions but overall great start.

great suggestions and make sense. We may need to couple of iterations to make it more solid.

File: docs/community/get_involved/developer_onboarding.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+receive emails and calendar invites send to Pyrsia Googlegroup.
+
+We collaborate on the project using Google Drive's shared directory
+[Pyrsia](https://drive.google.com/drive/folders/1vXO2YUjdyFXfjNGiKn82kUQ5ePrzp_jV?usp=share_link)
+
+Join Slack Workspace [Continuous Delivery Foundation](https://cdeliveryfdn.slack.com/). Join a few Slack channels #pyrsia, #pyrsia-team, #pyrsia-alert, #pyrsia-notifications
+
+Subscribe [PyrsiaOSS](https://www.youtube.com/channel/UClPQKloIElvJk7EdSST3W5g) Youtube channel for latest release, to know what is happening.
+
+## Technical Resources and Few Important Concepts
```
Comment (ihcomega56) [inline docs/community/get_involved/developer_onboarding.md:17]:
```suggestion
## Technical Resources and a Few Important Concepts
```

**Review by ihcomega56 — COMMENTED** [review]
@dasmanas  Can you fix the lint issue?
It is great to have this document in the web site.

**Review by betarelease — APPROVED** [review]

**Comment by dasmanas** [comment]
> 

solved.

**Review by ihcomega56 — APPROVED** [review]

**Review by betarelease — APPROVED** [review]

**Review by fishseabowl — APPROVED** [review]
lgtm

**Pull request merged**